Sources/QuranAudioKit/Preferences/AudioPreferences.swift

```diff
@@ -7,19 +7,50 @@
 
 import Combine
 import Foundation
+import OrderedCollections
 import Preferences
 
 @available(iOS 13.0, *)
 public class AudioPreferences {
     public static let shared = AudioPreferences()
+
     private init() {}
 
     private static let lastSelectedReciterId = PreferenceKey<Int>(key: "LastSelectedQariId", defaultValue: 41)
     private static let audioEndKey = PreferenceKey<Int>(key: "audioEndKey", defaultValue: AudioEnd.juz.rawValue)
+    private static let recentReciterIds = PreferenceKey<[Int]>(key: "recentRecitersIdsKey", defaultValue: [])
+    private static let recentReciterIdsTransfomer = PreferenceTransformer<[Int], OrderedSet<Int>>(
+        rawToValue: { OrderedSet($0) },
+        valueToRaw: { Array($0) }
+    )
+    private static let maxNumOfRecentReciters = 3
 
     @TransformedPreference(audioEndKey, transformer: .rawRepresentable(defaultValue: .juz))
     public var audioEnd: AudioEnd
 
     @Preference(lastSelectedReciterId)
     public var lastSelectedReciterId: Int
+
+    @TransformedPreference(recentReciterIds, transformer: recentReciterIdsTransfomer)
+    public var recentReciterIds: OrderedSet<Int>
+
+    public func recentReciters(_ allReciters: [Reciter]) -> [Reciter] {
+        var recentReciters: [Reciter] = []
+        for recentReciterId in self.recentReciterIds {
+            recentReciters.append(allReciters.filter { $0.id == recentReciterId }[0])
```

Comment:
Added a test case for the case when there are no recent reciters. It doesn't crash because it never goes inside the `for recentReciterId in self.recentReciterIds` loop (filter never gets called). 

This method is called by the App code (there is a corresponding PR in the private repo that's calling this method).

Reply:
In that case, write a test where self.recentReciterIds contains an id that is not present in allReciters - i.e. where filter will execute and return an empty array.

Reply:
Added the test for the edge case. It shouldn't be possible for a reciter ID that's not in allReciters to end up on the recentReciterIds but in case it happens, I've changed it to what you suggested so it doesn't crash.